### PR TITLE
Fix PTV-1717 / PUBLIC-1839

### PR DESCRIPTION
**Pull request by eapearson**
# Description of PR purpose/changes

Some modeling widgets, such as the FBAModel viewer, were reported to have broken landing page links. The user reported that the link included "functional-site" in the path, and that removing it resulted in a working link for the landing page. "functional-site" was the name of the precursor to kbase-ui, which was in use about 5 years ago ... so this link has been broken for a long time!

The fix was very simple - just an edit of one line. The overall change required more work, as the modeling widget source (kbase-extension/static/kbase/js/widgets/function_output/modeling) required some repair to be compliant with current practices in the codebase.

Two major flaws were left in the file, though, in order to avoid the need to refactor all of the modeling widgets - to wit, this code tends to use global modules, a practice which was superseded by AMD modules many years ago. The code does work, and converting all of the widgets to use or become good AMD citizens would require changing over 20 files, and involve hundreds if not thousands of edits.

Other caveats and notes:
- no additional tests were added; there is just one test for this file, loading the module; any functional tests would require much more work and surely some refactoring.
- a slightly updated package lock file was included, as the one in the develop branch is not quite updated with the state of package.json

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-X
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_
 - screenshots included

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [-] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [-] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [-] Any dependent changes have been merged and published in downstream modules
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook
- [-] (Python) I have run Black and Flake8 on changed Python code manually or with a git precommit hook

# Updating Version and Release Notes (if applicable)

- [-] [Version has been bumped](https://semver.org/) for each release
- [x] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

**Comment by eapearson** [comment]
Screen Shot 2021-11-22 at 6 12 28 PM

Screen Shot 2021-11-22 at 6 12 38 PM

Screen Shot 2021-11-22 at 6 12 49 PM

Screen Shot 2021-11-22 at 6 44 03 PM

File: kbase-extension/static/kbase/js/widgets/function_output/modeling/kbaseTabTable.js [hunk 1]
```diff
@@ -185,23 +182,23 @@ define([
...
                 };
 
                 // add any events
                 for (let i = 0; i < tab.columns.length; i++) {
-                    const col = tab.columns[i];
-
-                    settings.fnDrawCallback = function () {
-                        newTabEvents(tab.name);
-                    };
+                    settings.fnDrawCallback = fnDrawCallback;
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/function_output/modeling/kbaseTabTable.js:201]: what's the iterator being used for here?

File: kbase-extension/static/kbase/js/widgets/function_output/modeling/kbaseTabTable.js [hunk 2]
```diff
@@ -185,23 +182,23 @@ define([
...
-            this.getTableSettings = function (tab, data) {
+            this.getTableSettings = function (tab) {
                 const tableColumns = getColSettings(tab);
 
                 const settings = {
                     dom: '<"top"lf>rt<"bottom"ip><"clear">',
                     aaData: self.obj[tab.key],
                     aoColumns: tableColumns,
-                    language: { search: '_INPUT_', searchPlaceholder: 'Search ' + tab.name },
+                    language: {search: '_INPUT_', searchPlaceholder: 'Search ' + tab.name},
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/function_output/modeling/kbaseTabTable.js:192]: looks like this is a DataTables table -- can you update it to use the modern API? `aaData` and `aoColumns` are no longer valid -- I think the modern versions are `data` and `columns`, but you'd have to check.

File: kbase-extension/static/kbase/js/widgets/function_output/modeling/kbaseTabTable.js [hunk 3]
```diff
@@ -427,40 +424,27 @@ define([
...
-                return self.kbapi('biochem', 'get_reactions', { reactions: [id] }).then((data) => {
-                    return data[0];
-                });
-            };
-
-            this.getBiochemCompound = function (id) {
-                return self.kbapi('biochem', 'get_compounds', { compounds: [id] }).then((data) => {
-                    return data[0];
-                });
-            };
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/function_output/modeling/kbaseTabTable.js:460]: presumably these were never used

**Comment by ialarmedalien** [comment]
Could you fix the two problems shown in SonarCloud please?

File: .jshintrc [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "esversion": 6
```
[the remark this answers is not in the document]
Comment (briehl) [reply .jshintrc:2]: +1 to @ialarmedalien we're using eslint these days, no need to add another.

File: RELEASE_NOTES.md [hunk 5]
```diff
@@ -1,120 +1,136 @@
 ### OVERVIEW
 
-The Narrative Interface allows users to craft KBase Narratives using a combination of GUI-based commands, Python and R scripts, and graphical output elements.
+The Narrative Interface allows users to craft KBase Narratives using a combination of GUI-based commands, Python and R
```
Comment (briehl) [inline RELEASE_NOTES.md:3]: What happened here? Is there a markdown linter we should be using / enforcing now?

File: kbase-extension/static/kbase/js/widgets/function_output/modeling/kbaseTabTable.js (hunk 1 above)
Comment (briehl) [reply kbase-extension/static/kbase/js/widgets/function_output/modeling/kbaseTabTable.js:201]: That looks like maybe an old bug / old oversight. Part of what needs to be refactored here in this whole package. I think that's out of scope for now.

**Review by briehl — CHANGES_REQUESTED** [review]
I think the minor changes to kbaseTabTable are mostly fine. Just a couple questions.

File: .jshintrc (hunk 4 above)
[the remark this answers is not in the document]
Comment (eapearson) [reply .jshintrc:2]: This wasn't introduced for no reason, something in the development pipeline was unhappy without pinning the ecmascript version in jshint. Don't remember what it was now. Could just have been a local developer tool. Will remove and see if the issue crops up again.

File: kbase-extension/static/kbase/js/widgets/function_output/modeling/kbaseTabTable.js (hunk 3 above)
Comment (eapearson) [reply kbase-extension/static/kbase/js/widgets/function_output/modeling/kbaseTabTable.js:460]: correct

**Comment by eapearson** [comment]
@briehl @ialarmedalien I'm going to redo this PR from a new branch. The catchup merge from develop introduced so many files, which carried over 1K errors detected by the new husky pre-commit checks, I could not commit the merge without disabling such checks.
The new PR will have just two file changes - the same changes to kbaseTabTable.js, which is just one functional line change and a bunch of fixes to make the file commitable, and the release note.

**Comment by eapearson** [comment]
This has been superseded by PR # 2683

**Pull request closed without merging**